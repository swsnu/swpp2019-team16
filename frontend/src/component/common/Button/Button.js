import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

ButtonMaterial.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
  color: PropTypes.string,
  onClick: PropTypes.func,
  href: PropTypes.string,
  fullWidth: PropTypes.bool,
  height: PropTypes.string,
};

function ButtonMaterial({
  children,
  variant,
  color,
  onClick,
  href,
  fullWidth,
  height,
}) {
  return (
    <Button
      variant={variant}
      color={color}
      onClick={onClick}
      fullWidth={fullWidth}
      href={href}
      height={height}
    >
      {children}
    </Button>
  );
}

export default ButtonMaterial;
