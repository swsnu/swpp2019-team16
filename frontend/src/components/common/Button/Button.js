import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/styles';
import theme from '../../../lib/styles/theme';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    padding: '0 20px',
    fontSize: props => {
      switch (props.size) {
        case 'small':
          return '14px';
        case 'medium':
          return '13px';
        case 'large':
          return '15px';
        default:
          return '13px';
      }
    },
    height: props => {
      switch (props.size) {
        case 'small':
          return '28px';
        case 'medium':
          return '36px';
        case 'large':
          return '44px';
        default:
          return '36px';
      }
    },
    borderRadius: props => {
      switch (props.size) {
        case 'small':
        case 'medium':
        case 'large':
        default:
          return '4px';
      }
    },
  },
  contained: {
    boxShadow: props => (props.disableShadow ? 'none' : ''),
    border: props => (props.disableShadow ? '1px solid #E7E7E7' : 'none'),
    color: props => {
      switch (props.color) {
        case 'primary':
          return theme.palette.primary.contrastText;
        case 'secondary':
          return theme.palette.text.primary;
        default:
          return theme.palette.primary.contrastText;
      }
    },
    backgroundColor: props => {
      switch (props.color) {
        case 'primary':
          return theme.palette.primary.dark;
        case 'secondary':
          return theme.palette.primary.contrastText;
        default:
          return theme.palette.primary.dark;
      }
    },
  },
  text: {
    color: props => {
      switch (props.color) {
        case 'primary':
          return theme.palette.primary.contrastText;
        case 'secondary':
          return theme.palette.text.primary;
        default:
          return theme.palette.primary.contrastText;
      }
    },
  },
});

ButtonMaterial.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.oneOf(['contained', 'text']),
  color: PropTypes.oneOf(['primary', 'secondary']),
  backgroundColor: PropTypes.oneOf(['primary', 'secondary']),
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  href: PropTypes.string,
  fullWidth: PropTypes.bool,
};

function ButtonMaterial({
  children,
  variant = 'contained',
  color = 'primary',
  backgroundColor = 'primary',
  size = 'medium',
  disableShadow = false,
  fullWidth = false,
  disabled = false,
  onClick = () => {},
  href,
  endIcon,
}) {
  const styles = useStyles({
    size,
    color,
    backgroundColor,
    disableShadow,
    disabled,
  });
  return (
    <Button
      classes={{
        root: styles.root,
        contained: styles.contained,
        text: styles.text,
      }}
      disabled={disabled}
      variant={variant}
      onClick={onClick}
      fullWidth={fullWidth}
      href={href}
      endIcon={endIcon}
    >
      {children}
    </Button>
  );
}

export default ButtonMaterial;
