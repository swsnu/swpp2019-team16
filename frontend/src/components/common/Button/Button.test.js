import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('<Button />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = render(<Button children={<div>Button Text</div>} />);

    expect(container).toMatchSnapshot();
  });
});
