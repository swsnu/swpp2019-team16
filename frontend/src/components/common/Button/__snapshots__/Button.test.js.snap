// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> SHOULD match with snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <div>
        Button Text
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
